# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/vm/ArrayBufferViewObject.cpp
# Commit: 410e13fdf4e5
# Full Hash: 410e13fdf4e5927cf6ab312e77f00926a5669508
# Author: Steve Fink <sfink@mozilla.com>
# Date: 2021-08-13 09:27:46
# Regressor Bug: 1720422
# File Overlap Count: 1
# Description:
#   Bug 1720422 - Move typed array unwrap API to templatized version r=jonco
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D119856
# ==============================================================================

diff -r 677e4aa42b4d -r 410e13fdf4e5 js/src/vm/ArrayBufferViewObject.cpp
--- a/js/src/vm/ArrayBufferViewObject.cpp	Thu Aug 12 20:44:15 2021 +0000
+++ b/js/src/vm/ArrayBufferViewObject.cpp	Thu Aug 12 20:44:15 2021 +0000
@@ -241,6 +241,11 @@
   return length;
 }
 
+bool JS::ArrayBufferView::isDetached() const {
+  MOZ_ASSERT(obj);
+  return obj->as<ArrayBufferViewObject>().hasDetachedBuffer();
+}
+
 JS_PUBLIC_API size_t JS_GetArrayBufferViewByteOffset(JSObject* obj) {
   obj = obj->maybeUnwrapAs<ArrayBufferViewObject>();
   if (!obj) {